src/elf/sym.rs

```diff
@@ -186,20 +187,11 @@ macro_rules! elf_sym_std_impl {
                 }
             }
 
-            pub unsafe fn from_raw<'a>(symp: *const Sym, count: usize) -> &'a [Sym] {
-                slice::from_raw_parts(symp, count)
-            }
-
             pub fn from_fd<'a>(fd: &mut File, offset: usize, count: usize) -> Result<Vec<Sym>> {
                 // TODO: AFAIK this shouldn't work, since i pass in a byte size...
-                // FIX THIS, unecessary allocations + unsafety here
-                let mut bytes = vec![0u8; count * SIZEOF_SYM];
+                let mut syms = vec![Sym::default(); count];
                 try!(fd.seek(Start(offset as u64)));
-                try!(fd.read(&mut bytes));
-                let bytes = unsafe { slice::from_raw_parts(bytes.as_ptr() as *mut Sym, count) };
-                let mut syms = Vec::with_capacity(count);
-                syms.extend_from_slice(bytes);
-                syms.dedup();
```

Comment:
Haven't had a chance to give this a full review yet, but please don't remove API functions, e.g., this removal breaks dryad here and in 2 other places: https://github.com/m4b/dryad/blob/585189f3986c284aad52701d6de944a06192ab01/src/loader.rs#L197

Yes, I should probably write a safe wrapper for an elf symtable similar to [mach version](https://github.com/m4b/goblin/blob/8188a8a15017cbdadfdbb807e08db968f1c634c9/src/mach/symbols.rs#L164-L172),  and use this in clients, but don't have time for that right now (and its also a different PR).